ERCS/erc-7613.md

```diff
@@ -0,0 +1,345 @@
+// | GT             | 11            | [do not delegate] 0 0 32
+// Do not delegate if the first word of calldata is not a zero-padded address
+// | RETURNDATASIZE | 3D            | 0 [do not delegate] 0 0 32
+// | CALLDATALOAD   | 35            | [first word] [do not delegate] 0 0 32
+// | PUSH1          | 60 A0         | 160 [first word] [do not delegate] 0 0 32
+// | SHR            | 1C            | [first word upper bits] [do not delegate] 0 0 32
+// | OR             | 17            | [do not delegate] 0 0 32
+// Do not delegate if not called by the deployer
+// | PUSH20         | 73 [deployer] | [deployer] [do not delegate] 0 0 32
+// | CALLER         | 33            | [sender] [deployer] [do not delegate] 0 0 32
+// | XOR            | 18            | [sender not deployer] [do not delegate] 0 0 32
+// | OR             | 17            | [do not delegate] 0 0 32
+// Skip to the return if should not delegate
+// | PUSH1          | 60 40         | [success branch] [do not delegate] 0 0 32
+// | JUMPI          | 57            | 0 0 32
+// Calculate the payload size
+// | DUP3           | 82            | 32 0 0 32
+// | CALLDATASIZE   | 36            | [calldata size] 32 0 0 32
+// | SUB            | 03            | [payload size] 0 0 32
+// Copy the payload from calldata
+// | DUP1           | 80            | [payload size] [payload size] 0 0 32
+// | RETURNDATASIZE | 3D            | 0 [payload size] [payload size] 0 0 32
+// | SWAP5          | 94            | 32 [payload size] [payload size] 0 0 0
+// | RETURNDATASIZE | 3D            | 0 32 [payload size] [payload size] 0 0 0
+// | CALLDATACOPY   | 37            | [payload size] 0 0 0
+// Delegate call
+// | RETURNDATASIZE | 3D            | 0 [payload size] 0 0 0
+// | RETURNDATASIZE | 3D            | 0 0 [payload size] 0 0 0
+// | CALLDATALOAD   | 35            | [delegate to] 0 [payload size] 0 0 0
+// | GAS            | 5A            | [gas] [delegate to] 0 [payload size] 0 0 0
+// | DELEGATECALL   | F4            | [success] 0
+// Copy return data
+// | RETURNDATASIZE | 3D            | [return size] [success] 0
+// | DUP3           | 82            | 0 [return size] [success] 0
+// | DUP1           | 80            | 0 0 [return size] [success] 0
+// | RETURNDATACOPY | 3E            | [success] 0
+// Return
+// | SWAP1          | 90            | 0 [success]
+// | RETURNDATASIZE | 3D            | [return size] 0 [success]
+// | SWAP2          | 91            | [success] 0 [return size]
+// | PUSH1          | 60 40         | [success branch] [success] 0 [return size]
+// | JUMPI          | 57            | 0 [return size]
+// | REVERT         | FD            |
+// | JUMPDEST       | 5B            | 0 [return size]
+// | RETURN         | F3            |
+```
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
```

Comment:
Please move this paragraph to the top of your specification section (if you intend to use these keywords), or remove it entirely (if you don't.)

Reply:
Moved.